### ISSUE TITLE
Define allowed weapon anim type for companion

### ISSUE THREAD

**Issue by Lexx2k** [phobos2077/sfall#159]
Companion armor stuff has been moved to gl_npcarmor.ssl / npcarmor.ini while ago. It would be great to get an additional optional parameter to define which weapon animation the companion can use (Example: Spears, SMG, Pistol, etc).

Why?
It is possible to give any armor to companions, which then can result in a critter fid change, which then can result in the critter being able to wield more or less weapon types. This is inconsistent and can add to confusion for the player (why aren't you using your minigun?? ... while NPC has sulik look). Being able to define the allowed weapon types for a companion helps to mitigate this--- the companion will simply never wield a minigun, no matter if the current fid supports the animation or not.

**Comment by phobos2077** [comment]
You can code this in yourself, it's all in the script now. I think no sfall changes are required.

**Comment by Lexx2k** [comment]
Sure, but I'm not a good programmer / scripter and it probably would be a big mess that can be done much easier in ways I am unaware of. :>

**Comment by NovaRain** [comment]
I took some time to implement the idea.
Here's test version of it: http://www.mediafire.com/file/l4tws2ltku9kb3u/npcarmor_ext_nr.rar

I didn't commit to the repo yet because it's still kinda rough, my usual "quick-n-dirty" style.
Some notes about the script:
1. You have to set up a 'WeaponAnims' line in the INI to specify the weapon animation codes a NPC can use. For example:
```
; Sulik
[1]
PID=16777313
WeaponAnims=0,1,3,4,6
Default=16777280
...
```
The WeaponAnims line means Sulik will only equip weapons with 'none', 'knife', 'sledgehammer', 'spear', or 'SMG' anim code. NPCs will not equip any weapon if there's no WeaponAnims for them. Also if NPC is currently wielding a weapon they shouldn't use (e.g. Sulik in robes with a crowbar), clicking 'use best weapon' in combat preference will unwield the weapon.

2. I didn't check grenades. Because they use 'none' anim code, some NPCs might be able to equip them but can't really throw them. I was thinking to add a special number (like WeaponAnims=0,1,3,4,6,99) for such cases to do an extra check for weapon's projectile ID when anim code is 'none'. Or maybe some other way if anyone has better ideas.

**Comment by NovaRain** [comment]
OK, I tested a bit more and pushed to the repo. The script may not be the most elegant, but it should work fine.

**Comment by NovaRain** [comment]
So my implementation was bugged at the first place. While the bug got fixed, there's another issue surfaced during my testing:
If a party member has a weapon that his anim set allows to use but disallowed in the INI, and the weapon happens to be his "best weapon" (has highest AI priority), he won't equip other weapons and attack enemies in combat.
For example, Vic in power armor was able to equip .223 pistol. Now giving him both .223 pistol and assault rifle he won't equip anything because he always tries to equip .223 pistol but is prevented by the script.

I tried to make party members drop the weapon that they can but not allowed to equip, or move it back to player's inventory as a way to let them able to equip other weapons in the next time they try to equip something, but it's not reliable enough when in combat control panel (can work only once) and dropping/moving the weapon can still cause player confusion. IMO it seems like a bigger can of worms than the weapon use inconsistency/exploit. Maybe I should revert the change until there are better solutions.
